src/main/cli/cli.c

```diff
@@ -4091,7 +4091,21 @@ STATIC_UNIT_TESTED void cliSet(char *cmdline)
                     valueChanged = true;
 
                     break;
-
+                case MODE_STRING: {
+                        const unsigned int len = strlen(cmdline);
+                        const uint8_t min = val->config.string.minlength;
+                        const uint8_t max = val->config.string.maxlength;
+                        const bool updatable = ((val->config.string.flags & STRING_FLAGS_WRITEONCE) == 0 || strlen((char *)cliGetValuePointer(val)) == 0);
+
+                        if (updatable && len > 0 && len <= max) {
```

Comment:
In order to work for the `board_name` case, to be consistent with the required behaviour (stopping users from setting a config that is for a board that is different from what the config was created for), this should throw an error if `board_name` is set, and the new name does not match the old one.

Reply:
it will just not let you set it (throws an invalid error) once the initial setting has been made (regardless of what you set it to) - though re running a base config may be what you are referring to?

Reply:
Yes, or restoring a config from a backup - this was less of a problem before Unified Targets came along, but with a Unified Target, restoring a backup that was saved on a different board will make the board unusable.